server/src/main/kotlin/org/itmodreamteam/myrest/server/service/restaurant/RestaurantServiceImpl.kt

```diff
@@ -0,0 +1,69 @@
+package org.itmodreamteam.myrest.server.service.restaurant
+
+import org.itmodreamteam.myrest.server.error.UserException
+import org.itmodreamteam.myrest.server.model.restaurant.Restaurant
+import org.itmodreamteam.myrest.server.repository.restaurant.RestaurantRepository
+import org.itmodreamteam.myrest.shared.restaurant.RestaurantRegistrationInfo
+import org.itmodreamteam.myrest.shared.restaurant.RestaurantUpdateInfo
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+
+@Service
+class RestaurantServiceImpl (
+    private val restaurantRepository: RestaurantRepository,
+) : RestaurantService {
+
+    override fun register(newRestaurant: RestaurantRegistrationInfo) {
+        val existingRestaurant = restaurantRepository.findByName(newRestaurant.name)
+        if (existingRestaurant == null) {
+            restaurantRepository.save(Restaurant(newRestaurant))
+        } else {
+            throw UserException("Ресторан с таким именем уже существует")
+        }
+    }
+
+    override fun update(restaurantUpdateInfo: RestaurantUpdateInfo) {
+        if (restaurantUpdateInfo.name.isEmpty()) {
+            throw UserException("Неправильное имя ресторана")
+        }
+        if (restaurantUpdateInfo.getProperties().all { it.isNullOrBlank() }) {
```

Comment:
Would be clearer to move this check to RestaurantUpdateInfo class and name it like `containsUpdate() : Boolean`